app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -77,11 +46,31 @@ open class FeedViewModel(application: Application) : AndroidViewModel(applicatio
     fun getNetworkErrorEvent(): LiveData<Event<String>> = networkErrorEvent
 
     init {
-        refresh()
+        val disposableA =  feedRepository.getAllProjects().subscribe({ userProjects ->
+            updateState {
+                State(
+                    userProjects.map { userProject ->
+                        FeedItem(userProject.id, userProject.thumbnailUrl, userProject.isPremium)
+                    },
+                    false
+                )
+            }
+        }, { error ->
+            updateState { copy(getFeedItems().value, false) }
```

Comment:
When using `copy()` you only need to specify the properties that changed.
Did you feed items change because of the error?